hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/ChecksumFileSystem.java

```diff
@@ -670,6 +670,27 @@ public boolean rename(Path src, Path dst) throws IOException {
     }
   }
 
+  /**
+   * Rename files/dirs.
+   */
+  @Override
+  @SuppressWarnings("deprecation")
+  public void rename(final Path src,
+      Path dst,
+      final Options.Rename... options)
+      throws IOException {
+      fs.rename(src, dst, options);
```

Comment:
The deprecated version checks the return code here and bails if the rename wasn't successful. If Options includes Options.Rename.NONE and dst already exists, this will return false? If so, we shouldn't proceed to overwrite the checksums.

Reply:
if options = NONE and the dest exists, the call MUST raise an exception. I've just reviewed this carefully and added comments to clarify